client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -107,6 +111,22 @@ const CheckoutThankYou = React.createClass( {
 		return getPurchases( props ).some( purchase => isPlan( purchase ) || isDomainProduct( purchase ) );
 	},
 
+	renderConfirmationNotice: function() {
+		if ( ! this.props.user || ! this.props.user.email || this.props.email_verified ) {
+			return;
```

Comment:
I think it's preferable to return `null` or `false` if we don't want to render anything. This returns `undefined` instead. See https://github.com/facebook/react/issues/3738.